packages/apollo-server-core/src/plugin/landingPage/default/index.ts

```diff
@@ -142,6 +142,7 @@ id="embeddableSandbox"
     target: '#embeddableSandbox',
     initialEndpoint,
     includeCookies: ${config.includeCookies ?? 'false'},
+    initialState: ${config},
```

Comment:
`initialState` should only have 3 properties, maybe better to be explicit about passing only what it expects?

```suggestion
    initialState: ${{
      document: config.document ?? undefined,
      variables: config.variables ?? undefined,
      headers: config.headers ?? undefined,
    }},
```

Reply:
@William010x and I identified that you can't just drop an object into a string template like this (you get an `[object Object]`), so it's now `JSON.stringify()`ed.

Additionally, I'll mention that the type of `initialState` is now a `string` and it'll need to be `JSON.parse()`d in the constructor to be useful.

Reply:
With the substitutions in the template string (`${JSON.stringify(expression)}`), it actually removes the quotes surrounding the JSON object string, so the returned HTML string will look something like:

```
<script>
  var initialEndpoint = window.location.href;
  new window.EmbeddedSandbox({
    target: '#embeddableSandbox',
    initialEndpoint,
    includeCookies: false,
    initialState: {"document":"query test { id }","variables":{"v":"1"},"headers":{"h":"2"}},
  });
</script>
```

I tested this HTML string with the Embedded Sandbox and it seems to work without needing to `JSON.parse()`.



Alternatively, doing something like these (similar to the embeddedExplorerHTML) also works
```
<script>
  var initialEndpoint = window.location.href;
  var state = ${JSON.stringify({
      document: config.document ?? undefined,
      variables: config.variables ?? undefined,
      headers: config.headers ?? undefined,
    })};
  new window.EmbeddedSandbox({
    target: '#embeddableSandbox',
    initialEndpoint,
    includeCookies: ${config.includeCookies ?? 'false'},
    initialState: state,
  });
</script>
```

```
<script>
  var initialEndpoint = window.location.href;
  var state = ${JSON.stringify({
      document: config.document ?? undefined,
      variables: config.variables ?? undefined,
      headers: config.headers ?? undefined,
    })};
  new window.EmbeddedSandbox({
    target: '#embeddableSandbox',
    initialEndpoint,
    includeCookies: ${config.includeCookies ?? 'false'},
    initialState: {
        ...state
    },
  });
</script>
```

Reply:
I'll look into adding proper unit tests for this though!

Reply:
Neat, yeah if it's not adding quotes like I expected then you should be all set!